gcloud/bigquery/dataset.py

```diff
@@ -0,0 +1,339 @@
+        """
+        client = self._require_client(client)
+
+        api_response = client.connection.api_request(
+            method='GET', path=self.path)
+        self._set_properties(api_response)
+
+    def patch(self, client=None, **kw):
+        """API call:  update individual dataset properties via a PATCH request
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/v2/datasets/patch
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+
+        :type kw: ``dict``
+        :param kw: properties to be patched.
+        """
+        client = self._require_client(client)
+
+        partial = {}
+
+        if 'default_table_expiration_ms' in kw:
+            value = kw['default_table_expiration_ms']
+            if not isinstance(value, int) and value is not None:
+                raise ValueError()
+            partial['defaultTableExpirationMs'] = value
+
+        if 'description' in kw:
+            partial['description'] = kw['description']
+
+        if 'friendly_name' in kw:
+            partial['friendlyName'] = kw['friendly_name']
+
+        if 'location' in kw:
+            partial['location'] = kw['location']
+
+        api_response = client.connection.api_request(
+            method='PATCH', path=self.path, data=partial)
+        self._set_properties(api_response)
+
+    def update(self, client=None):
+        """API call:  update dataset properties via a PUT request
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/v2/datasets/update
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+        """
+        client = self._require_client(client)
+        api_response = client.connection.api_request(
+            method='PUT', path=self.path, data=self._build_resource())
+        self._set_properties(api_response)
+
+    def delete(self, client=None):
+        """API call:  delete the dataset via a DELETE request
+
+        See:
+        https://cloud.google.com/bigquery/reference/rest/v2/datasets/delete
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+        """
+        client = self._require_client(client)
+        client.connection.api_request(method='DELETE', path=self.path)
+
+
+def _datetime_from_prop(value):
+    """Convert non-none timestamp to datetime, assuming UTC.
+
+    :rtype: ``datetime.datetime``, or ``NoneType``
+    """
+    if value is not None:
+        value = datetime.datetime.utcfromtimestamp(value / 1000.0)
```

Comment:
https://cloud.google.com/bigquery/docs/reference/v2/datasets#resource says:

> creationTime 	long 	[Output-only] The time when this dataset was created, in milliseconds since the epoch.

which we preserve:

```python
>>> from gcloud.bigquery.client import Client
>>> client = Client()
>>> ds = client.dataset('simple_test')
>>> ds.create()
>>> ds.created
datetime.datetime(2015, 7, 27, 17, 36, 43, 913000, tzinfo=<UTC>)
>>> ds.modified
datetime.datetime(2015, 7, 27, 17, 36, 43, 913000, tzinfo=<UTC>)
>>> ds.delete()
```